domain-models-runtime/src/main/java/org/folio/rest/persist/ddlgen/SchemaMaker.java

```diff
@@ -122,6 +126,49 @@ private String handleDelete() throws IOException, TemplateException {
     return writer.toString();
   }
 
+  /**
+   * @return the tables of schema plus tables to delete (tables that exist in previousSchema but not in schema);
+   *   add foreign keys to delete (those that exist in previousSchema but not in schema).
+   *   Nothing to do for indexes because rmb_internal_index handles them.
+   */
+  List<Table> tables() {
+    if (previousSchema == null || previousSchema.getTables() == null) {
+      return schema.getTables();
+    }
+    Map<String,Table> tableForName = new HashMap<>();
+    schema.getTables().forEach(table -> tableForName.put(table.getTableName(), table));
+    List<Table> list = new ArrayList<>(schema.getTables());
+    previousSchema.getTables().forEach(oldTable -> {
+      Table newTable = tableForName.get(oldTable.getTableName());
+      if (newTable == null) {
+        oldTable.setMode("delete");
+        oldTable.setup();
+        list.add(oldTable);
+        return;
+      }
+
+      List<ForeignKeys> oldForeignKeys = oldTable.getForeignKeys();
+      if (oldForeignKeys == null || oldForeignKeys.isEmpty()) {
+        return;
+      }
+      List<ForeignKeys> newForeignKeys =
+          newTable.getForeignKeys() == null ? Collections.emptyList() : newTable.getForeignKeys();
+      List<ForeignKeys> allForeignKeys = new ArrayList<>(newForeignKeys);
+      oldForeignKeys.forEach(oldForeignKey -> {
+        if (newForeignKeys.stream().anyMatch(newForeignKey ->
+            oldForeignKey.getFieldName().equals(newForeignKey.getFieldName()))) {
```

Comment:
@adamdickmeiss @julianladisch oldForeignKey.getFieldName() causes NPE in case of upgrading tables which contains cross-index foreign keys without field names like
`    {
      "tableName": "titles",
      "fromModuleVersion": "mod-orders-storage-11.0.0",
      "withMetadata": true,
      "foreignKeys": [
        {
          "fieldName": "poLineId",
          "targetTable":      "po_line",
          "targetTableAlias": "poLine",
          "tableAlias": "titles",
          "tOps": "ADD"
        },
        {
          "targetPath": ["poLineId", "purchaseOrderId"],
          "targetTable":      "purchase_order",
          "targetTableAlias": "purchaseOrder",
          "tableAlias": "titles",
          "tOps": "ADD"
        }
      ]
    },`